test/e2e/autoscaling/dns_autoscaling.go

```diff
@@ -65,10 +70,20 @@ var _ = SIGDescribe("DNS horizontal autoscaling", func() {
 		nodeCount := len(nodes.Items)
 
 		ginkgo.By("Collecting original replicas count and DNS scaling params")
+
+		// Check if we are running coredns or kube-dns, the only difference is the name of the autoscaling CM.
+		// The test should be have identically on both dns providers
+		provider, err := detectDNSProvider(ctx, c)
+		framework.ExpectNoError(err)
+
 		originDNSReplicasCount, err = getDNSReplicas(ctx, c)
 		framework.ExpectNoError(err)
+		configMapNames = map[string]string{
+			"kube-dns": "kube-dns-autoscaler",
+			"coredns":  "coredns-autoscaler",
```

Comment:
where are these names defined?

Reply:
Over here:
https://github.com/kubernetes/kops/blob/63c69fe9dd83446f9c415bf5c6249ffe0b7d2fcf/upup/models/cloudup/resources/addons/coredns.addons.k8s.io/k8s-1.12.yaml.template#L349
https://github.com/kubernetes/kubernetes/blob/0dc900cebe079efaf2087adc6fee6eb88d176020/cluster/addons/dns-horizontal-autoscaler/dns-horizontal-autoscaler.yaml#L97

These are the configmaps the autoscaler creates

Reply:
There are only 2 DNS providers. Kubedns is deprecated, I'm happy to make this support only coredns and lose test coverage for kubedns clusters.

Reply:
we can not loose coverage

Reply:
Both scenarios will still be tested. There is a CI job that runs with kubedns as the DNS provider instead of coredns. It picked up the buggy change I did last time